tokio/src/runtime/handle.rs

```diff
@@ -144,6 +144,10 @@ impl Handle {
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
+        if self.is_shutdown() {
+            return JoinHandle::new_dropped_runtime();
+        }
+
         #[cfg(all(tokio_unstable, feature = "tracing"))]
         let future = crate::util::trace::task(future, "task");
         self.spawner.spawn(future)
```

Comment:
What if the runtime is shut down right after the `self.is_shutdown()` call?

Reply:
Good catch. We should probably test for the shutdown after the `self.spawner.spawn` call.